helix-term/src/ui/picker.rs

```diff
@@ -123,70 +124,120 @@ impl Preview<'_, '_> {
     }
 }
 
-fn item_to_nucleo<T: Item>(item: T, editor_data: &T::Data) -> Option<(T, Utf32String)> {
-    let row = item.format(editor_data);
-    let mut cells = row.cells.iter();
-    let mut text = String::with_capacity(row.cell_text().map(|cell| cell.len()).sum());
-    let cell = cells.next()?;
-    if let Some(cell) = cell.content.lines.first() {
-        for span in &cell.0 {
-            text.push_str(&span.content);
-        }
-    }
-
-    for cell in cells {
-        text.push(' ');
-        if let Some(cell) = cell.content.lines.first() {
-            for span in &cell.0 {
-                text.push_str(&span.content);
-            }
+fn inject_nucleo_item<T, D>(
+    injector: &nucleo::Injector<T>,
+    columns: &[Column<T, D>],
+    item: T,
+    editor_data: &D,
+) {
+    let column_texts: Vec<Utf32String> = columns
```

Comment:
I believe the collect is necessary because of value taking the `T` but the column_texts taking `&T`. There are some borrow conflicts for:

```rust
let column_texts = columns
    .iter()
    .filter(|column| column.filter)
    .map(|column| Utf32String::from(column.format_text(&item, editor_data)));

injector.push(item, |dst| {
    for (dst, text) in dst.iter_mut().zip(column_texts) {
        *dst = text;
    }
});
```

but I might be misunderstanding your suggestion here

Reply:
Ag yeah I think I see the problem now. That is an issue with nucleos API imo, I will look into improving this upstream

Reply:
I released a new nucleo version to address the problem and a pushed a commit that avoids collecting to a temporary vec